rssbot/tasks.py

```diff
@@ -30,31 +32,39 @@ def rssbot_task(app):
                 if last_updated and last_updated == feed_last_updated:
                     continue
                 else:
-                    for item in parsed_feed["entries"]:
+                    for entry in parsed_feed["entries"]:
                         exists = await Outbox.find_one({
                             "user_id": bot.name,
-                            "feed_item_id": item["id"]
+                            "feed_item_id": entry["id"]
                         })
                         if exists:
                             continue
                         else:
-                            content = item.get("summary", None) or item.get("content", None)[0]["value"]
-                            if not (content and bot["details"]["rssbot"]["html"]):
-                                content = item['title']
-
                             body_tags = ""
                             object_tags = []
+                            body_tags_list = []
+                            tag_list = []
+
+                            content = entry.get("summary", None) or entry.get("content", None)[0]["value"]
+                            if not (content and bot["details"]["rssbot"]["html"]):
+                                content = entry['title']
+
+                            find_tag_scheme = r"(?!<a[^>]*?>)(?P<tagged>#\w+)(?![^<]*?</a>)"
+                            find_tag_scheme = re.compile(find_tag_scheme)
+                            text_tags = re.findall(find_tag_scheme, content)
+                            if text_tags:
```

Comment:
I suggest next solution. Creating 2 lists, one "footer_tags" (ex body_tags_list), another "apub_tags" (apub means activitypub in short, ex object_tags). Then extract entry tags, as "entry_tags"
`footer_tags = set(entry_tags + bot["details"]["rssbot"]["tags"])`
then update footer_tags with octothorpe
`apub_tags = set(footer_tags + text_tags)`
And then convert these lists to the target representation

Reply:
ok